$OpenBSD: patch-src_options_c,v 1.5 2016/01/13 12:39:55 gsoares Exp $

use SSLv23_client_method() required to build with libressl since that it haven't
TLS_client_method()  for now.

--- src/options.c.orig	Tue Jan  5 06:20:12 2016
+++ src/options.c	Tue Jan 12 19:57:30 2016
@@ -2559,7 +2559,7 @@ NOEXPORT char *parse_service_option(CMD cmd, SERVICE_O
     /* sslVersion */
     switch(cmd) {
     case CMD_BEGIN:
-#if OPENSSL_VERSION_NUMBER>=0x10100000L
+#if OPENSSL_VERSION_NUMBER>=0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)
         section->client_method=(SSL_METHOD *)TLS_client_method();
         section->server_method=(SSL_METHOD *)TLS_server_method();
 #else
@@ -2571,7 +2571,7 @@ NOEXPORT char *parse_service_option(CMD cmd, SERVICE_O
         if(strcasecmp(opt, "sslVersion"))
             break;
         if(!strcasecmp(arg, "all")) {
-#if OPENSSL_VERSION_NUMBER>=0x10100000L
+#if OPENSSL_VERSION_NUMBER>=0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)
             section->client_method=(SSL_METHOD *)TLS_client_method();
             section->server_method=(SSL_METHOD *)TLS_server_method();
 #else
